### DISCUSSION TITLE
Download pdf from yaml url

### DISCUSSION THREAD

**Discussion by S1M0N38** [papis/papis#525] [Ideas]
In the documentation, I cannot find a command to download a PDF using the URL contained in info.yaml.

My workflow is as follows:

1. I use papis add `--from arxiv https://...` to add a paper to my library and download the PDF locally.
2. I store my library in a Git repository and push it to GitHub, but I exclude the PDF files by adding *.pdf to .gitignore. This means that only the YAML files are pushed.
3. Another user can clone my repository and use the URL contained in the YAML files to download the PDF to their local machine.

I think that this is an effective way to maintain a shared library with collaborators using version control.

However, there is currently no simple command to download a PDF using the URL contained in info.yaml (e.g. `papis download [QUERY]`).

Currently, my collaborators can use `papis browser [QUERY]` to inspect the PDF in their browser, but it would be nice for them to have the PDF locally.

Of course, I could write a custom script, but I wonder if something like this is already implemented or if there are plans to implement it.

>If the paper is not on arXiv, I add it using `--from doi`. Then I search for a public PDF version and add it to the library with `papis addto -f my_paper.pdf`. Finally, I edit the URL contained in the corresponding YAML file to point to the URL where I can download my_paper.pdf.

**Comment by alexfikl** [comment]
I don't think there's a direct command to do that at the moment, but it should be a nice addition to `papis addto`, i.e. something like
```
papis addto --from arxiv '...' query
```
similar to how `papis update` and `papis add` work.
